src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -214,3 +214,18 @@ def random_string(length=16, force_lower=False, digits_only=False):
     if not digits_only:
         choice_set += ascii_lowercase if force_lower else ascii_letters
     return ''.join([choice(choice_set) for _ in range(length)])
+
+
+def hash_string(value, length=16, force_lower=False):
```

Comment:
In all of the application of this method below, I didn't see in any case the length is beyond 64, actually none of them uses a name string longer than 16. Therefore, why not just limit the return length to 64. When `length` is greater than 64, which is the sha256 hex digest length, simply throw an exception. By doing so, you reduce the complexity of this method and eliminating the unused behavior.